test/getGrist.ts

```diff
@@ -220,12 +259,33 @@ export class GristUtils extends GristWebDriverUtils {
 
   // Crude, assumes a single iframe. Should elaborate.
   public async getCustomWidgetBody(selector: string = 'html'): Promise<string> {
-    const iframe = driver.find('iframe');
+    const iframe = this.driver.find('iframe');
     try {
       await this.driver.switchTo().frame(iframe);
-      return await driver.find(selector).getText();
+      return await this.driver.find(selector).getText();
     } finally {
-      await driver.switchTo().defaultContent();
+      await this.driver.switchTo().defaultContent();
+    }
+  }
+
+  public async executeInIframe(fnc: (driver: WebDriver) => void) {
+    const iframe = this.driver.find('iframe');
+    try {
+      await this.driver.switchTo().frame(iframe);
+      await fnc(this.driver);
+    } finally {
+      await this.driver.switchTo().defaultContent();
+    }
+  }
+
+  public async getCustomWidgetObject(script: string): Promise<any> {
```

Comment:
This is a wrapper around `executeScript`. If you look at use of `executeScript` in grist-core, it is usually passed a function rather than a string, which is a lot neater. I'm not exactly sure how this works though.